pkg/store/instance.go

```diff
@@ -250,31 +264,96 @@ func PrintInstances(w io.Writer, instances []*Instance, format string) error {
 		}
 		fmt.Fprintln(w, string(b))
 	case "table":
+		types := map[string]int{}
+		archs := map[string]int{}
+		for _, instance := range instances {
+			types[instance.VMType]++
+			archs[instance.Arch]++
+		}
+		all := options != nil && options.AllFields
+		width := 0
+		if options != nil && options.IsTerminal {
+			width = options.TerminalWidth
+		}
+		hideType := false
+		hideArch := false
+		hideDir := false
+
+		columns := 1 // NAME
+		columns += 2 // STATUS
+		columns += 2 // SSH
+		if width != 0 && (columns+7)*8 > width && !all {
+			hideType = len(types) == 1
+		}
+		if !hideType {
+			columns++ // VMTYPE
+		}
+		hideArch = len(archs) == 1 && !all
```

Comment:
We only want to hide the ARCH column if all instances are using the host arch

```suggestion
		hideArch = len(archs) == 1 && instances[0].Arch == runtime.GOARCH && !all
```

Reply:
I think strictly speaking that the runtime.GOARCH is the binary arch and not the host arch

At least we have had some qemu/rosetta incidents in the path, so maybe it needs a function ?

Reply:
> I think strictly speaking that the runtime.GOARCH is the binary arch and not the host arch

Yes, but we use it everywhere else as the host arch (e.g. in the `lima.yaml` defaults), so it would still be consistent.

> At least we have had some qemu/rosetta incidents in the path, so maybe it needs a function ?

Does this really happen? If you use the `x86_64` version of `limactl` on `aarch64` with Rosetta2, it would also try to call the `qemu-system-x86_64` binary, which most likely won't be installed.

I think the code I suggested is good enough, and matches what we do in other places to determine the host arch.